marklogic-data-hub/src/main/java/com/marklogic/hub/dhs/installer/command/InstallIntoDhsCommand.java

```diff
@@ -29,22 +32,34 @@
     public void run(ApplicationContext context, Options options) {
         initializeProject(context, options, buildDefaultProjectProperties(options));
 
-        logger.info("Installing DHF version " + hubConfig.getJarVersion());
+        Versions versions = new Versions(hubConfig);
+        String installedHubVersion = versions.getInstalledVersion();
+        MarkLogicVersion mlVersion = versions.getMLVersion(serverVersion);
 
-        HubAppDeployer deployer = new HubAppDeployer(hubConfig.getManageClient(), hubConfig.getAdminManager(), null, null);
+        ObjectNode canInstall = canInstallDhs(installedHubVersion, mlVersion);
+        if(canInstall.get("canBeInstalled").asBoolean()){
+            logger.info("Installing DHF version " + hubConfig.getJarVersion());
 
-        String groupName = "Evaluator";
-        modifyHubConfigForDhs(groupName);
-        deployer.setCommands(buildCommandsForDhs(options));
-        deployer.deploy(hubConfig.getAppConfig());
+            HubAppDeployer deployer = new HubAppDeployer(hubConfig.getManageClient(), hubConfig.getAdminManager(), null, null);
 
-        // Update the servers in the Curator group
-        groupName = "Curator";
-        modifyHubConfigForDhs(groupName);
-        DhsDeployServersCommand dhsDeployServersCommand = new DhsDeployServersCommand();
-        dhsDeployServersCommand.setServerVersion(serverVersion);
-        deployer.setCommands(Arrays.asList(dhsDeployServersCommand));
-        deployer.deploy(hubConfig.getAppConfig());
+            String groupName = "Evaluator";
+            modifyHubConfigForDhs(groupName);
+            deployer.setCommands(buildCommandsForDhs(options));
+            deployer.deploy(hubConfig.getAppConfig());
+
+            // Update the servers in the Curator group
+            groupName = "Curator";
+            modifyHubConfigForDhs(groupName);
+            DhsDeployServersCommand dhsDeployServersCommand = new DhsDeployServersCommand();
+            dhsDeployServersCommand.setServerVersion(serverVersion);
+            deployer.setCommands(Arrays.asList(dhsDeployServersCommand));
+            deployer.deploy(hubConfig.getAppConfig());
+        }
+        else {
+            logger.error("Failed to install DHF version " + hubConfig.getJarVersion());
```

Comment:
I think this message should be "Unable to install ", as a failure hasn't occurred yet.